connector/datadogconnector/connector.go

```diff
@@ -80,6 +93,9 @@ func getTraceAgentCfg(cfg TracesConfig, attributesTranslator *attributes.Transla
 	acfg.ComputeStatsBySpanKind = cfg.ComputeStatsBySpanKind
 	acfg.PeerTagsAggregation = cfg.PeerTagsAggregation
 	acfg.PeerTags = cfg.PeerTags
+	if len(cfg.ResourceAttributesAsContainerTags) > 0 {
+		acfg.Features["enable_cid_stats"] = struct{}{}
```

Comment:
There is another relevant feature flag `disable_cid_stats` https://github.com/DataDog/datadog-agent/blob/b7ad1ad429262138b4bd198d48874616d98b7417/pkg/trace/stats/concentrator.go#L200

To be safe you can add `delete(acfg.Features, "disable_cid_stats")` too